### ISSUE TITLE
dplyr 1.0

### ISSUE THREAD

**Issue by bpbond** [JGCRI/gcamdata#1153]
As noted by @ashiklom on Slack, 

>Heads up: dplyr v1.0.0 release is on the horizon. Lots of new features, plus a fair amount of breaking changes. Biggest change I can see is that all of the *_if, *_at , and *_all functions are being deprecated in favor of a new across function that, AFAICT, looks something like this: iris %>% summarize(across(starts_with("Sepal"), ~mean(x, na.rm = TRUE)) . Might not be a bad idea to start testing code (in particular, gcamdata ) against the development version of dplyr to keep on top of possible breaking changes.

https://github.com/tidyverse/dplyr/blob/master/NEWS.md

**Comment by pralitp** [comment]
And indeed it breaks functionality in the gcamdata system.  The first error a user would encounter is:
```
1] "module_aglu_LA100.0_LDS_preprocessing"
Error: L100.Land_type_area_ha is being returned grouped. This is not allowed; please ungroup()
```

**Comment by jedwardsmdy** [comment]
whats the last version of dplyr that it does work for?  I just ran it with dplyr 0.8.0, dplyr 0.8.3, dplyr 0.8.5 and it throws an error:

`[1]` "module_aglu_LA100.IMAGE_downscale_ctry_yr"
`Error` in mutate_impl(.data, dots, caller_env()) :  attempt to bind a variable to R_UnboundValue

current version of tidyr requires dplyr 0.8.0 or higher so can't go any farther back with out using older versions of other packages.  this is run with R 4.0.2 don't know if that matters.

**Comment by pralitp** [comment]
There have been so many of these tidy verse updates that break compatibility it is hard to keep track.  From the log of the successful CI run on the last commit:
```
remotes::install_deps(dependencies = TRUE, type="win.binary")'
gcamdata.... (NA -> c03483a54...) [GitHub]
assertthat   (NA -> 0.2.1       ) [CRAN]
dplyr        (NA -> 0.8.4       ) [CRAN]
magrittr     (NA -> 1.5         ) [CRAN]
tibble       (NA -> 2.1.3       ) [CRAN]
tidyr        (NA -> 1.0.2       ) [CRAN]
readr        (NA -> 1.3.1       ) [CRAN]
data.table   (NA -> 1.12.8      ) [CRAN]
igraph       (NA -> 1.2.4.2     ) [CRAN]
mockr        (NA -> 0.1         ) [CRAN]
testthat     (NA -> 2.3.1       ) [CRAN]
R.utils      (NA -> 2.9.2       ) [CRAN]
usethis      (NA -> 1.5.1       ) [CRAN]
rlang        (NA -> 0.4.4       ) [CRAN]
cli          (NA -> 2.0.1       ) [CRAN]
crayon       (NA -> 1.3.4       ) [CRAN]
fansi        (NA -> 0.4.1       ) [CRAN]
pillar       (NA -> 1.4.3       ) [CRAN]
pkgconfig    (NA -> 2.0.3       ) [CRAN]
glue         (NA -> 1.3.1       ) [CRAN]
utf8         (NA -> 1.1.4       ) [CRAN]
vctrs        (NA -> 0.2.3       ) [CRAN]
ellipsis     (NA -> 0.3.0       ) [CRAN]
digest       (NA -> 0.6.24      ) [CRAN]
purrr        (NA -> 0.3.3       ) [CRAN]
R6           (NA -> 2.4.1       ) [CRAN]
Rcpp         (NA -> 1.0.3       ) [CRAN]
tidyselect   (NA -> 1.0.0       ) [CRAN]
BH           (NA -> 1.72.0-3    ) [CRAN]
plogr        (NA -> 0.2.0       ) [CRAN]
desc         (NA -> 1.2.0       ) [CRAN]
pkgbuild     (NA -> 1.0.6       ) [CRAN]
rprojroot    (NA -> 1.3-2       ) [CRAN]
rstudioapi   (NA -> 0.11        ) [CRAN]
withr        (NA -> 2.1.2       ) [CRAN]
callr        (NA -> 3.4.2       ) [CRAN]
prettyunits  (NA -> 1.1.1       ) [CRAN]
backports    (NA -> 1.1.5       ) [CRAN]
processx     (NA -> 3.4.2       ) [CRAN]
ps           (NA -> 1.3.2       ) [CRAN]
R.methodsS3  (NA -> 1.8.0       ) [CRAN]
ini          (NA -> 0.3.1       ) [CRAN]
jsonlite     (NA -> 1.6.1       ) [CRAN]
httr         (NA -> 1.4.1       ) [CRAN]
curl         (NA -> 4.3         ) [CRAN]
mime         (NA -> 0.9         ) [CRAN]
openssl      (NA -> 1.4.1       ) [CRAN]
askpass      (NA -> 1.1         ) [CRAN]
sys          (NA -> 3.3         ) [CRAN]
stringi      (NA -> 1.4.6       ) [CRAN]
lifecycle    (NA -> 0.1.0       ) [CRAN]
hms          (NA -> 0.5.3       ) [CRAN]
clipr        (NA -> 0.7.0       ) [CRAN]
lazyeval     (NA -> 0.2.2       ) [CRAN]
evaluate     (NA -> 0.14        ) [CRAN]
pkgload      (NA -> 1.0.2       ) [CRAN]
praise       (NA -> 1.0.0       ) [CRAN]
R.oo         (NA -> 1.23.0      ) [CRAN]
clisymbols   (NA -> 1.2.0       ) [CRAN]
fs           (NA -> 1.3.1       ) [CRAN]
gh           (NA -> 1.1.0       ) [CRAN]
git2r        (NA -> 0.26.1      ) [CRAN]
whisker      (NA -> 0.4         ) [CRAN]
yaml         (NA -> 2.2.1       ) [CRAN]
```

That being said, I don't think I have seen that particular error message so it could be R 4 related (the last CI run used R 3.6).

We were able to work through the dplyr 1.0 fixes, which also is being tested on R 4.0, and get everything running successfully (hopefully we will get that reviewed and available soon).  However even there I didn't seen any changes directly affecting `module_aglu_LA100.IMAGE_downscale_ctry_yr` so I am not sure..

**Comment by aliciaszhao** [comment]
> whats the last version of dplyr that it does work for? I just ran it with dplyr 0.8.0, dplyr 0.8.3, dplyr 0.8.5 and it throws an error:
> 
> ```
> [1] "module_aglu_LA100.IMAGE_downscale_ctry_yr"
> Error in mutate_impl(.data, dots, caller_env()) :  attempt to bind a variable to R_UnboundValue
> ```
> 
> current version of tidyr requires dplyr 0.8.0 or higher so can't go any farther back with out using older versions of other packages. this is run with R 4.0.2 and most recent versions of other dependencies.

I'm having this same issue, even after changing the package versions to the ones indicated above.

**Comment by Silviameteoro** [comment]
Hi,

I am having a similar issue with dplyr1.0.0. You mentioned that you could run the driver with R 3.6 but later you said you were having this same issue. Then I got confused if this is working for you or not... Thanks

**Comment by aliciaszhao** [comment]
> Hi,
> 
> I am having a similar issue with dplyr1.0.0. You mentioned that you could run the driver with R 3.6 but later you said you were having this same issue. Then I got confused if this is working for you or not... Thanks

Switching to R 3.6 resolved my issue with dplyr!

**Comment by pralitp** [comment]
closed in #1161
